### ISSUE TITLE
Steam friends closing steam.

### ISSUE THREAD

**Issue by lordSeaworth** [awesomeWM/awesome#2021]
Output of `awesome --version`:
4.2-1

Just wondering if there is any workaround or fix for Steam friends where it happens to close steam down after closing the friends window. I can't close it without shutting down Steam.

Any suggestions or a direction where to look would be nice.

**Comment by 0x041E** [comment]
Unable to reproduce for me works correctly closing friends window either by close on awesome's titlebar of the window, steam friends window's native close button or by key bind, all of these close only the friends window. To me it seems like steam issue. Are you sure that steam is gracefully closed when you close friends window and not crashing? If you start steam through terminal steam will print logs to the terminal so you can check if it's getting closed or it's crashing on closing the friends window. Alternatively could you try reproducing the bug under different window manager to check it's awesome wm specific problem?

**Comment by Elv13** [comment]
Reference suggested by `folk` on IRC

https://github.com/ValveSoftware/steam-for-linux/issues/322

**Comment by blueyed** [comment]
That issue is from 2013 however?!

**Comment by kajzersoze** [comment]
Issue still exists, I've noticed it doesn't matter if it's friends window , chat window or whatever window, as long as you close it via titlebar close button you will close the entire Steam client, if you close it with native Steam button it's ok.

**Comment by psychon** [comment]
Two random thoughts:
- Anything on steam's stdout/stderr?
- Does the friends window support the `WM_DELETE_WINDOW` protocol? (Run `xprop WM_PROTOCOLS`, click the window, check if `WM_DELETE_WINDOW` appears in the output)

**Comment by kajzersoze** [comment]
* Nothing on steam's stdout, it acts like it would when you want to close the Steam client (it just says Shutdown)
* It does support it :  `WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW`
* Tried only I3wm and it works there.

**Comment by psychon** [comment]
> Maybe that's what's making the difference here, because if here in Awesome I close the friends window with key bind then it works (doesn't close the entire client) , but if I close it with close button on titlebar then it does bring the whole client down.

Huh?!?!? That shouldn't make any difference, both cases send a `WM_DELETE_WINDOW` message to the window in question. No idea what is going on here.

Only idea that I have has: Can you try running steam under `x11trace` (or `xtrace` if that's what your distro calls it)?
Basically:

- Run `x11trace --outfile /tmp/log.txt --relative-timestamps steam`
- Open the steam friends window
- Wait 10 seconds (so that I can use the timestamps to find the interesting part of the log file)
- Close the friends window via some keyboard combination
- Wait 10 seconds
- Open the steam friends window
- Wait 10 seconds
- Close the friends window via the titlebar

Alternatively, you can also use `--timestamps` instead of `--relative-timestamps`. This then prints the wall clock time to the log file (e.g. 10:17:42). If you would then write down the exact times that you closed the windows (but still wait a bit around these actions!), that would likely make it even simpler to understand the log file.

Afterwards, provide `/tmp/log.txt` (but beware, it might get huge; possibly (g)zip it?; and in theory it should be possible to piece together steam's window content and keyboard input from this; basically: this is more powerful than a keylogger) and the times. I'll take a look at it (as time permits).

**Comment by kajzersoze** [comment]
Well, it is what it is, it shouldn't make any difference but it does and that's what I'm experiencing.

I'll try to do xtrace later.

**Comment by kajzersoze** [comment]
Have some problems with xtrace (never used it before) :
```
$ xtrace --outfile /tmp/log.txt --relative-timestamps steam                                                                                                                                                                    
xtrace: unrecognized option `--outfile'
Try `xtrace --help' or `xtrace --usage' for more information.
```
```
$ xtrace --help                                                                                                                                                                                                                
Usage: xtrace [OPTION]... PROGRAM [PROGRAMOPTION]...
Trace execution of program by printing currently executed function.

     --data=FILE          Don't run the program, just print the data from FILE.

   -?,--help              Print this help and exit
      --usage             Give a short usage message
   -V,--version           Print version information and exit

Mandatory arguments to long options are also mandatory for any corresponding
short options.
```
```
$ xtrace --usage                                                                                                                                                                                                              
Usage: xtrace [OPTION]... PROGRAM [PROGRAMOPTION]...
```
I mean it doesn't matter (I don't even use titlebars), but I tested it and noticed that problem, I can reproduce it, if OP doesn't care anymore and Awesome devs don't have Steam installed then let's leave this open until some better times.

**Comment by psychon** [comment]
@kajzersoze Try installing and using x11trace. There is a (useless?) utility called xtrace that is part of glibc, which is why some distros call this x11trace instead. Sorry if I wasn't clear enough about this in my previous comment. https://xtrace.alioth.debian.org/

**Comment by kajzersoze** [comment]
Ok, I did install it and that part is working now, so I enabled titlebars and proceeded with your instructions, funny thing .... now it doesn't happen anymore, when I close the Friends window via titlebar it just closes Friends window. 
I'm not sure what's going on but I swear it did happen couple of days ago.
Well, never mind my previous report I guess.

**Comment by ryukinix** [comment]
Funny problem. I can replicate that too.  But only close the steam when I close via awesome, not the `x` close button of friends window.

**Comment by lordSeaworth** [comment]
Well it has been a while but for me the steam closes on using the awesome drawn titlesbars still applies. But you can 'fix' it by just adding the 
`{ rule = { class = "Steam" },
    properties = { titlebars_enabled = false } },` to your rc.lua so it doesn't draw the titlesbars so you won't accidently click the awesome titlebar and close steam.

**Comment by ryukinix** [comment]
> Well it has been a while but for me the steam closes on using the awesome drawn titlesbars still applies. But you can 'fix' it by just adding the
{ rule = { class = "Steam" }, properties = { titlebars_enabled = false } }, to your rc.lua so it doesn't draw the titlesbars so you won't accidently click the awesome titlebar and close steam.

This can be true, but for instance in my use-case I disable globally titlebars. But even disabled If I close then window by `mod+c` this still close the whole application. :/

**Comment by kajzersoze** [comment]
Yeah still same, if closed by bind key (mod+q in my case) entire steam client closes.
Didn't try with titlebar enabled what happens since my last report here, but when I tried I'm sure it was closing steam, next time it didn't, so it's weird.
Anyway, it closes with key bind always, and it shouldn't, so maybe solving that one would solve the titlebar issue as well.
